### PR TITLE
LP-2382 Implement Github actions pipeline for linting

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

[LP-2382](https://philanthropyu.atlassian.net/browse/LP-2382) Creating Github actions workflow for pylint and iSort

### Notes

- Once this PR is merged into develop, any PR made to be merged in develop branch will trigger this Github action (build).
- Any push or update in that PR will also trigger Github action (build).
- Same is the case for master, when this PR will merge in master.
- Only python files in PR will be checked, in Github action.
- [ ] Core changes will be logged, once PR is reviewed

Improvements for future:

- Add cache for OS dependencies and Django requirements
- Update bot messaging and apply tools only to git diff lines instead of git diff files

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: .github/workflows/quality.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+name: Code quality with Pylint & iSort
+
+on:
+  # Trigger workflow on pull request, but only for PRs merging into zahid/LP-2361-from-dev branch
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - master
```
Comment (emraan004) [inline .github/workflows/quality.yml:8]: do we need to run linter on master branch when it has already run on develop branch?

Comment (zahid-arbisoft) [reply .github/workflows/quality.yml:8]: We can skip it for master branch, if you want.

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Review by saad-waqar — APPROVED** [review]

File: .github/workflows/quality.yml (hunk 1 above)
Comment (ahmed-faraz46) [reply .github/workflows/quality.yml:8]: @zahid-arbisoft @emraan004 did we decide it?

File: .github/workflows/quality.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  # Trigger workflow on pull request, but only for PRs merging into master and develop branches
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - master
+      - develop
+
+jobs:
+  quality:
+
```
Comment (ahmed-faraz46) [inline .github/workflows/quality.yml:13]: Can we please remove this new line?

File: .github/workflows/quality.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    - name: Install edx-platform requirements
+      run: |
+        pip install -r requirements/edx/testing.txt
+        pip install -r requirements/philu/testing.txt
+
+    - name: Analysing code with iSort
+      id: isort-report
+      continue-on-error: true
+      run: |
+        isort ${{ env.GIT_DIFF }} --diff 2>&1 |tee report_isort.txt
```
Comment (ahmed-faraz46) [inline .github/workflows/quality.yml:43]: Let's please revisit the indentation, May be a space after `>` and `|` is required.

File: .github/workflows/quality.yml (hunk 1 above)
Comment (emraan004) [reply .github/workflows/quality.yml:8]: @ahmed-faraz46 going with both branches

**Review by emraan004 — APPROVED** [review]
:

File: philu_actions/scripts/dependency_downloader.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -xe
+
+SHORT_DIST="xenial" # Update this if you update ubuntu version
+
+# Bootstrapping constants
+PIP_VERSION="9.0.3"
+EDX_PPA_KEY_SERVER="keyserver.ubuntu.com"
+EDX_PPA_KEY_ID="B41E5E3969464050"
```
Comment (ahmed-faraz46) [inline philu_actions/scripts/dependency_downloader.sh:10]: Is this something public?

File: philu_actions/scripts/dependency_downloader.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
```
Comment (ahmed-faraz46) [inline philu_actions/scripts/dependency_downloader.sh:1]: If we haven't already, may we check with @NIXKnight? there might be some more sophisticated way to install edx system dependencies, written in ancible playbooks or configurations?

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]
Great work @zahid-arbisoft 👏 here are a few things I think we can revisit, rest of the PR looks really great to me 👍

File: .github/workflows/quality.yml (hunk 3 above)
Comment (zahid-arbisoft) [reply .github/workflows/quality.yml:43]: Added space after `|`, however `2>&1` is normally written [as is](https://unix.stackexchange.com/questions/20469/difference-between-21-output-log-and-21-tee-output-log).

File: philu_actions/scripts/dependency_downloader.sh (hunk 4 above)
Comment (zahid-arbisoft) [reply philu_actions/scripts/dependency_downloader.sh:10]: Yes, it is taken from [edx ansible script](https://github.com/edx/configuration/blob/08800400716e918f9b08c6ee05e24f734a798104/util/install/ansible-bootstrap.sh#L47rL48)

File: philu_actions/scripts/dependency_downloader.sh (hunk 5 above)
Comment (zahid-arbisoft) [reply philu_actions/scripts/dependency_downloader.sh:1]: I have taken this script from one of the [ansible script by Edx](https://raw.githubusercontent.com/edx/configuration/master/util/install/ansible-bootstrap.sh), and customized it as per our (& Github actions) requirements

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Review by ahmed-faraz46 — APPROVED** [review]
LGTM 👍

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Pull request merged**